main/src/ca/uwaterloo/flix/language/phase/TypeSimplifier.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2025 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase
+
+import ca.uwaterloo.flix.language.ast.{Kind, Type}
+import ca.uwaterloo.flix.language.phase.typer.TypeConstraint
+import ca.uwaterloo.flix.language.phase.unification.EffUnification3
+import ca.uwaterloo.flix.util.InternalCompilerException
+
+
+/** This type simplification is purely focused on user readability. */
+object TypeSimplifier {
+
+  /**
+    * Simplifies types and effects - intended for user readability.
+    *
+    * This function will never crash and works for any type, no matter how ill-kinded.
+    */
+  def simplify(tpe: Type): Type = {
+    try {
+      simplifyInternal(tpe)
+    } catch {
+      case _: InternalCompilerException => tpe
```

Comment:
This is not acceptable. You can catch a more precise exception, but you cannot catch ICE

Reply:
You also shouldn't crash when trying to display an error - should i just remove the catch?